src/components/Button/ConnectButton.tsx

```diff
@@ -80,7 +87,12 @@ const ConnectButton: React.FC<Props> = ({ className = 'blue' }) => {
 	}
 
 	const withoutWeb3 = async () => {
-		setProfile({ address: ethers.constants.AddressZero })
+		const sessionID = await getOffChainSession()
+		setProfile({
+			address: ethers.constants.AddressZero,
+			isOffChain: true,
```

Comment:
Thoughts on making address optional and using that nullability as the offChain flag? Setting address to `address(0)` was a bit of a hack.

Reply:
sweet, done